### PR TITLE
osd: small clear up and optimization

### PR DESCRIPTION
**Pull request by songbaisen**
osd: small clear up and optimization

Signed-off-by: song baisen <song.baisen@zte.com.cn>

### CODE REVIEW

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -980,6 +981,7 @@ bool OSDService::should_share_map(entity_name_t name, Connection *con,
                << " has old map " << epoch << " < "
                << osdmap->get_epoch() << dendl;
       should_send = true;
+      return should_send;
```
Comment (branch-predictor) [inline src/osd/OSD.cc]: This doesn't really optimize anything, only makes code less readable.

File: src/osd/OSD.cc [hunk 2]
```diff
@@ -1058,6 +1060,7 @@ bool OSDService::can_inc_scrubs_pending()
     dout(20) << __func__ << " " << scrubs_pending << " -> " << (scrubs_pending+1)
 	     << " (max " << cct->_conf->osd_max_scrubs << ", active " << scrubs_active << ")" << dendl;
     can_inc = true;
+    return can_inc;
```
Comment (branch-predictor) [inline src/osd/OSD.cc]: Ditto.

File: src/osd/OSD.cc [hunk 3]
```diff
@@ -1068,17 +1071,17 @@ bool OSDService::can_inc_scrubs_pending()
 bool OSDService::inc_scrubs_pending()
 {
   bool result = false;
+  Mutex::Locker l(sched_scrub_lock);
```
Comment (branch-predictor) [inline src/osd/OSD.cc]: Using RAII is safer but costs (extra object construction and destruction).

File: src/osd/OSD.cc [hunk 4]
```diff
@@ -1068,17 +1071,17 @@ bool OSDService::can_inc_scrubs_pending()
...
 
-  sched_scrub_lock.Lock();
   if (scrubs_pending + scrubs_active < cct->_conf->osd_max_scrubs) {
     dout(20) << "inc_scrubs_pending " << scrubs_pending << " -> " << (scrubs_pending+1)
 	     << " (max " << cct->_conf->osd_max_scrubs << ", active " << scrubs_active << ")" << dendl;
     result = true;
     ++scrubs_pending;
+    return result;
```
Comment (branch-predictor) [inline src/osd/OSD.cc]: Like above, this doesn't optimize anything.

File: src/osd/OSD.cc (hunk 1 above)
Comment (songbaisen) [inline src/osd/OSD.cc]: @branch-predictor  Yes that is true.

File: src/osd/OSD.cc [hunk 5]
```diff
@@ -962,6 +962,7 @@ bool OSDService::should_share_map(entity_name_t name, Connection *con,
                << " versus osdmap epoch " << osdmap->get_epoch() << dendl;
       if (*sent_epoch_p < osdmap->get_epoch()) {
         should_send = true;
+        return should_send;
```
Comment (songbaisen) [inline src/osd/OSD.cc]: @branch-predictor  How about this, good eyes man. Here we can quickly return.

File: src/osd/OSD.cc (hunk 3 above)
Comment (songbaisen) [inline src/osd/OSD.cc]: @branch-predictor  Oh you are right! Good!

**Pull request closed without merging**